Sources/Sentry/Public/SentryHub.h

```diff
@@ -55,8 +57,26 @@ SENTRY_NO_INIT
                                  operation:(NSString *)operation
     NS_SWIFT_NAME(startTransaction(name:operation:));
 
+/**
+ * Creates a transaction bound it to the hub and returns the instance.
+ * The created transaction is added to the scope using the given key.
+ * A transaction already in the scope with the same key is replaced.
+ *
+ * @param name The transaction name.
+ * @param operation Short code identifying the type of operation the span is measuring.
+ * @param key A key to bound the transaction to the scope.
+ *
+ * @return The created transaction.
+ */
+- (id<SentrySpan>)startTransactionWithName:(NSString *)name
+                                 operation:(NSString *)operation
+                                  scopeKey:(NSString *)key
```

Comment:
why don't we use the transaction's name? they are unique anyway

Reply:
This is an overload. The method without the scopeKey parameter uses the name as key.
How can transaction name be unique if its user defined? I understand that this is the best scenario. But the user can call startTransaction twice with the same name, Gods know why. Each project has its particularities.

Reply:
@brustolin they can, but our server groups by name, if they have the same name and they are not the same thing, wrong things will be grouped, it's misusing the API, it's specified in the docs https://docs.sentry.io/product/performance/

Reply:
Oh, yes, the server side of things I understand.
But the user needs ways to handle the transactions in the app. 

Think this way: I have two separated UIViews, each one loads a list of News from distinct categories. Each view starts a transaction with the name "LoadNews" but set an extra value in the context with the category id.

After the load is finish the user needs to retrieve the transaction from the scope to also finish the transaction.
In this case the user can use the CategoryId as key, but in Sentry this transaction should be grouped because they do the same thing.

Reply:
@brustolin yes, this is a known problem, my comment here is having an overload with an extra parameter `scopeKey ` instead of using the `name` directly

Reply:
It's good that you think about such edge cases @brustolin. IMO we should just go with using the transaction name as a key to simplify the API. If a user has such an edge case he can also create a map himself to keep track of his transactions. I think we should reflect in the API how the server is handling transactions. For me, the extra key argument makes the API complex and confusing.